rustls-libssl/tests/server.c

```diff
@@ -0,0 +1,185 @@
+/**
+ * Simple server test program.
+ *
+ * Listens on the given port, and accepts one connection on it.
+ *
+ */
+
+#include <assert.h>
+#include <netdb.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include <openssl/x509.h>
+
+static int trace(int rc, const char *str) {
```

Comment:
now that there's two test C programs (`server.c` and `client.c`) WDYT about doing something similar to rustls-ffi and lifting out a `common.c` for these helpers? (`trace`, `TRACE`, `hexdump`, and `dump_openssl_error_stack`).

Reply:
Added a commit that extracts the common parts, just into a header for now.

Reply:
Perfect, thanks!